src/game/Coordinate.java

```diff
@@ -80,7 +80,7 @@ private boolean isVertical(Coordinate other) {
 	public boolean equals(Object obj) {
 		if (obj instanceof Coordinate) {
 			Coordinate other = (Coordinate) obj;
-			return this.valid && other.isValid() && this.x == other.getX() && this.y == other.getY();
+			return this.valid == other.isValid() && this.x == other.getX() && this.y == other.getY();
```

Comment:
changed this so that when i stress test random coordinates, it doesn't create like 100,000 buckets